core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -110,7 +110,8 @@ class ReplicaManager(val config: KafkaConfig,
   /* epoch of the controller that last changed the leader */
   @volatile var controllerEpoch: Int = KafkaController.InitialControllerEpoch - 1
   private val localBrokerId = config.brokerId
-  private val allPartitions = new Pool[(String, Int), Partition]
+  private val partitionFactory = (k: (String, Int)) => new Partition(k._1, k._2, time, this)
+  private val allPartitions = new Pool[(String, Int), Partition](Some(partitionFactory))
```

Comment:
Could we not simply inline the factory since it's not used anywhere else? Something like the following maybe?

```scala
private val allPartitions = new Pool[(String, Int), Partition](valueFactory = Some { case (t, p) =>
  new Partition(t, p, time, this)
})
```